static_inverse_kinematics.py

```diff
@@ -219,3 +234,21 @@ def animate(self):
         b.load_experimental_markers(self.xp_markers)
         b.load_movement(self.q)
         b.exec()
+
+    def get_sol(self):
+        for i, sol in enumerate(self.sol):
+            for j, value in enumerate(sol.fun):
+                self.residuals_xyz[i][j] = value
+            self.nb_iteration_diff.append(sol.nfev)
+            self.nb_iteration_jac.append(sol.njev)
+
+        for j, value in enumerate(self.residuals_xyz):
+            for h in range(0, 3 * self.nb_markers, 3):
+                self.residuals[j][int(h / 3)] = np.linalg.norm(value[h : h + 3])
+
+        self.output = dict(
+            time=self.duration,
+            residuals=self.residuals,
+            nb_iteration_diff=self.nb_iteration_diff,
+            nb_iteration_jac=self.nb_iteration_jac,
+        )
```

Comment:
All the variables stored in output should not be attributes of the method.

replace   `self.residuals_xyz[i][j]` by  `residuals_xyz[i][j]`
Same for every other variables.

You should iterate on the `self.nb_frame` instead of the `self.sol`
Try to find a better one for `sol.fun`

To be more generic, you can create a new attribute `self.n_dim` related to the dimension of markers, we could have 2D motion capture so it would be:
`self.nb_dims * self.nb_markers` instead of `*3`

Anyway, your loop needs to be clarified.
Please start your outer loop with the number of markers.
There should be only one loop.

FInal comments, don't use nested list, use numpy arrays for example `self.residuals[j,int(h / 3)]`.

Reply:
self.nb_dim done
iterate on self.nb_frames done 
replace self.sol by self.list_sol done

Reply:
self.name_of_variables as been change to name_of_variables